src/scheduler/Scheduler.cpp

```diff
@@ -193,31 +185,19 @@ void Scheduler::notifyFaasletFinished(const faabric::Message& msg)
     thisHostResources.set_boundexecutors(newBoundExecutors);
 }
 
-std::shared_ptr<InMemoryMessageQueue> Scheduler::getBindQueue()
-{
-    return bindQueue;
-}
-
-Scheduler& getScheduler()
-{
-    // Note that this ref is shared between all faaslets on the given host
-    static Scheduler scheduler;
-    return scheduler;
-}
-
 std::vector<std::string> Scheduler::callFunctions(
```

Comment:
The logic the `callFunctions` is the heart of the scheduler and was a bit sketchy, so it's undergone quite a bit tidy-up. Best to look at the full thing: https://github.com/faasm/faabric/blob/batch-execution/src/scheduler/Scheduler.cpp#L188